### DISCUSSION TITLE
[@types/gsi] bizarre typing of `google.accounts.id.renderButton`

### DISCUSSION THREAD

**Discussion by drewboardman** [DefinitelyTyped/DefinitelyTyped#63692] [Issues with a @types package]
I'm new to both TS and React - and I'm running into an issue. I'm trying to use the Google Sign In API to send google's JWT to my webserver for verification.

the problem is that I'm seeing a discrepancy in what the google official documentation suggests, and the definitely-typed package that exists for `gsi` (google sign in). I'm not familiar enough with the ecosystem to know if this package is just wrong - or if I'm misunderstanding something.

Here is the specific issue:

```
getElementById(elementId: string): HTMLElement | null;
function renderButton(parent: string, options: GsiButtonConfiguration): void;
```

[documentation](https://developers.google.com/identity/gsi/web/guides/display-button#javascript) says:

```
google.accounts.id.renderButton(
            document.getElementById("buttonDiv"),
            { theme: "outline", size: "large" }  // customization attributes
          );
```

This doesn't type check - since the `getElementById` is returning the whole element. This makes me have to guess at what the implementation is asking me to pass to `renderButton`

The Mozilla documentation seems to suggest that this is indeed not a `string`:

> The Document method getElementById() returns an Element object representing the element whose id property matches the specified string. Since element IDs are required to be unique if specified, they're a useful way to get access to a specific element quickly.

[a comment by an automated account is not shown]

**Reply by drewboardman** [reply]
It seems like the most recent version on `master` fixes this - but npm has a stale version?

**Reply by aronhoyer** [reply]
i don't see anything in master that fixes your reported issue. but in response to your report, yes, the type of the parent param of renderButton is incorrect. i can take a look at it later today unless someone else picks this up 👍

**Comment by drewboardman** [comment]
It seems like the @types/google.accounts is a correct version of these
types.

**Reply by aronhoyer** [reply]
i don't have any familiarity with `@types/google.accounts`, but it doesn't seem like it types the same api as `@types/gsi` does

**Comment by drewboardman** [comment]
Seems to be working for me, same use case as gsi 🤷‍♂️
